courses/serializers/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""Course serializer shared functions"""
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.templatetags.static import static
+from rest_framework import serializers
+
+from courses import models
+
+
+def get_thumbnail_url(page):
+    """
+    Get the thumbnail URL or else return a default image URL.
+
+    Args:
+        page (cms.models.ProductPage): A product page
+
+    Returns:
+        str:
+            A page URL
+    """
+    relative_url = (
+        page.thumbnail_image.file.url
+        if page
+        and page.thumbnail_image
+        and page.thumbnail_image.file
+        and page.thumbnail_image.file.url
+        else static("images/mit-dome.png")
+    )
+    return urljoin(settings.SITE_BASE_URL, relative_url)
+
+
+class BaseProgramRequirementTreeSerializer(serializers.ListSerializer):
```

Comment:
We talked about this (very) briefly in Slack but I'm not sure this goes here now? Since we're not using it in v2 and we're leaving the v1 stuff as-is (so v2 is just deltas) I think this is best kept alongside the v1 serializers that use it. My thinking is that if we need a v2 API for this then we'll probably also have a use case for refactoring the `ProgramRequirementTreeSerializer` and/or this serializer as well. 

(The only thing that uses this, I think, is the `ProgramAdminForm` - it's the only place right now that needs a fully realized requirements tree. If we expose this in the frontend later, we will have some additional requirements for that at that point in time.)

What do you think?

Reply:
So, it is subclassed by the `[ProgramRequirementTreeSerializer](https://github.com/mitodl/mitxonline/pull/1966/files#diff-551e72a56450d28fbe49c33b7992eccf1b5b990308a7198a77cec2bc1726b2c3R94)` - while there's no real change, I think that was the reason I pulled it up. I can drop it if you don't think it's useful to have at the root, I just didn't want to import from v1 here.

Reply:
Ok, yeah.. I think my brain went splat on this again. I think this is OK as is in this case.